### PR TITLE
ELPP-3128 Temporary failure on CDN Invalidation

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

File: tests/activity/test_activity_invalidate_cdn.py [hunk 1]
```diff
@@ -24,11 +25,29 @@ def test_invalidation_success(self, fake_emit, invalidation_mock):
...
+        self.assertEqual(result, self.invalidatecdn.ACTIVITY_TEMPORARY_FAILURE)
+
+    @patch('time.sleep')
+    @patch('provider.cloudfront_provider.create_invalidation')
+    @patch.object(activity_InvalidateCdn, 'emit_monitor_event')
+    def test_invalidation_temporary_failure_fixed_backoff(self, fake_emit, invalidation_mock, sleep_mock):
+        invalidation_mock.side_effect = CloudFrontServerError(400, 'Bad Request', '<ErrorResponse xmlns="http://cloudfront.amazonaws.com/doc/2010-11-01/"><Error><Type>Sender</Type><Code>TooManyInvalidationsInProgress</Code><Message>...</Message></Error><RequestId>...</RequestId></ErrorResponse>')
+        result = self.invalidatecdn.do_activity(activity_data)
+        self.assertEqual(result, self.invalidatecdn.ACTIVITY_TEMPORARY_FAILURE)
+        self.assertEqual(sleep_mock.mock_calls, [call(60)])
```
Comment (gnott) [inline tests/activity/test_activity_invalidate_cdn.py]: I was confused where the call(60) comes from, I see now the sleep 60 seconds is in the activity itself, and you're not relying on the default 10 second wait between failed activity attempts in the workflow.py object?

Comment (giorgiosironi) [reply tests/activity/test_activity_invalidate_cdn.py]: I took this from https://github.com/elifesciences/elife-bot/blob/db0462fcc31b2cf218fc198d904ca004c2962ead/activity/activity_VerifyGlencoe.py#L78
but if it's dealt with at the workflow level it's even better. 10 seconds would definitely be enough, I didn't want it to retry continuously.

File: tests/activity/test_activity_invalidate_cdn.py [hunk 2]
```diff
@@ -24,11 +25,29 @@ def test_invalidation_success(self, fake_emit, invalidation_mock):
...
+    @patch.object(activity_InvalidateCdn, 'emit_monitor_event')
+    def test_invalidation_temporary_failure_due_to_rate_limiting(self, fake_emit, invalidation_mock, sleep_mock):
+        invalidation_mock.side_effect = CloudFrontServerError(400, 'Bad Request', '<ErrorResponse xmlns="http://cloudfront.amazonaws.com/doc/2010-11-01/"><Error><Type>Sender</Type><Code>TooManyInvalidationsInProgress</Code><Message>Processing your request will cause you to exceed the maximum number of in-progress wildcard invalidations.</Message></Error><RequestId>b47186e6-b7d7-11e7-bd6c-d7fe045b879d</RequestId></ErrorResponse>')
+        result = self.invalidatecdn.do_activity(activity_data)
+        self.assertEqual(result, self.invalidatecdn.ACTIVITY_TEMPORARY_FAILURE)
+
+    @patch('time.sleep')
+    @patch('provider.cloudfront_provider.create_invalidation')
+    @patch.object(activity_InvalidateCdn, 'emit_monitor_event')
+    def test_invalidation_temporary_failure_fixed_backoff(self, fake_emit, invalidation_mock, sleep_mock):
```
[the remark this answers is not in the document]
Comment (giorgiosironi) [reply tests/activity/test_activity_invalidate_cdn.py]: Two different tests on the same scenario:
- that we decide to raise a temporary failure
- the time we spend backing off before retrying
However with the previous changes the second test will vanish

**Comment by giorgiosironi** [comment]
Updated simplifying the sleep.

**Comment by thewilkybarkid** [comment]
Can this be merged? Seeing regular end2end failures.

**Comment by thewilkybarkid** [comment]
Not just regular now, happens every time when trying to update Journal.

File: activity/activity_InvalidateCdn.py [hunk 3]
```diff
@@ -40,8 +41,22 @@ def do_activity(self, data):
...
+                if e.error_code == 'TooManyInvalidationsInProgress':
+                    self.logger.warning(e)
+                    self.emit_monitor_event(
+                        self.settings,
+                        article_id, 
+                        version,
+                        run,
+                        self.pretty_name,
+                        "error",
+                        "Glencoe video is not available for article " + article_id + '; message: ' + str(e)
```
Comment (thewilkybarkid) [inline activity/activity_InvalidateCdn.py]: Message doesn't sound right?

**Pull request merged**